### PR TITLE
REQUEST FOR CODE REVIEW ON JAVASCRIPT TO-DO LIST PROJECT

### PR DESCRIPTION
**Pull request by Moreneecoder**
In this project, I built a to-do app that allows users to manage tasks and sort them into projects. The project makes use of Javascript and the Webpack bundler. I also modularized the code to enable a DRY and clean code pattern. Some of the features built include:

* A landing page in the Home tab
* A default Welcome project with a task that says 'Start here' in the App tab
* A new task button to create new tasks
* Each task has an edit and delete button
* User can click on task checkbox to mark it has done
* The edit button and delete button will change to 'done' when a task is checked
* When creating a task, users can create a new project in the process.
* The new project will be added to the select project options when created.

### CODE REVIEW

File: src/components/appCompontents/app.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import * as Projects from './factories/project';
+
+const loadProjects = (projects) => {
+  let list = '';
+  projects.forEach((project) => {
```
Comment (elisha2kyakpo1) [inline src/components/appCompontents/app.js]: - [ ] I was unable to create a new project, the project requirement is that a user should be able to create a new project

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    //   }
+
+    //   if (e.target && e.target.id === 'contact') {
+    //     domActions.removeChildWithinParent(mainContent, ['menu-wrapper', 'home-wrapper']);
+    //     domActions.addTabContent(mainContent, 'contact-wrapper', contactComponent());
+    //   }
+  });
+});
+
+//   document.body.appendChild(component());
```
Comment (elisha2kyakpo1) [inline src/index.js]: - [ ] please, remove this commented code, remember to leave the codebase clean, if you need any of the old code, you can use GitHub history.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi, @Moreneecoder, 

Good job so far! I like the project setup as well as the codebase.
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  I wasn't able to delete or edit a to-do item, this is a requirement for this project.
- [ ] When I clicked the new task button, the popup opens, but I can't create anything from this popup.
- [ ] The default projects could not allow me to create a to-do from them, they are static.

_Check other comments under the review._ 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


### Cheers and Happy coding!👏👏👏

**__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by taiwo2 — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED ♻️ 
@Moreneecoder well done on this project work and thanks for implementing some of the suggested changes, however, changes are required
### Highlight

- I was able to delete or edit a to-do item, this is a requirement for this project. 👍🏻 
-  When I clicked the new task button, the popup opens, I was able to create from the popup. 👍🏻

 
### Improve 

- [ ] The default projects should be static i.e should not be edited or delete or create a to-do from them. they should serve as example for users

### once fixed kindly request for review

**Comment by Moreneecoder** [comment]
The last two reviewers have conflicting opinions on the default project. The first, Elisha, says the default project should not be static, as he expects a user to be able to create and modify todos from them (the default welcome project). The second reviewer, Taiwo, said totally otherwise. This is seriously confusing and tiring because it is the only aspect of this project holding me from approval. I think the first reviewer is right, however, because, I don't see why the default project should be static. Other apps like Todoist, don't have a static default project. It's totally dynamic.

**Review by afanimartin — APPROVED** [review]
## Approved 👍🏿 👍🏿 👍🏿 
Sorry for the confusion. I think from the requirements we are only expected to have a default project. Merge and proceed to the next project.


### Martin Afani
#### Slack: whiz25

**Pull request merged**